protocols/perf/Cargo.toml

```diff
@@ -0,0 +1,40 @@
+[package]
+name = "libp2p-perf"
+edition = "2021"
+rust-version = "1.64.0"
+description = "libp2p perf protocol implementation"
+version = "0.1.0"
+authors = ["Max Inden <mail@max-inden.de>"]
+license = "MIT"
+repository = "https://github.com/libp2p/rust-libp2p"
+keywords = ["peer-to-peer", "libp2p", "networking"]
+categories = ["network-programming", "asynchronous"]
+
+[dependencies]
+anyhow = "1"
+async-std = { version = "1.9.0", features = ["attributes"] }
+clap = { version = "4.1.6", features = ["derive"] }
+env_logger = "0.10.0"
+futures = "0.3.26"
+instant = "0.1.11"
+libp2p-core = { version = "0.39.0", path = "../../core" }
+libp2p-dns = { version = "0.39.0", path = "../../transports/dns", features = ["async-std"] }
+libp2p-noise = { version = "0.42.0", path = "../../transports/noise" }
+libp2p-quic = { version = "0.7.0-alpha.2", path = "../../transports/quic", features = ["async-std"] }
+libp2p-swarm = { version = "0.42.0", path = "../../swarm", features = ["macros", "async-std"] }
+libp2p-tcp = { version = "0.39.0", path = "../../transports/tcp", features = ["async-io"] }
+libp2p-yamux = { version = "0.43.0", path = "../../muxers/yamux" }
```

Comment:
We only need these for the binaries right? How about we introduce a separate crate that implements the server and client binaries, similarly to `interop-tests`? That would allow this protocol to be WASM-friendly. Plus, in the spirit of https://github.com/libp2p/rust-libp2p/issues/3111, it would be consistent with not introduce binaries in our protocol crates.

Reply:
Which folder should these binaries go into?

> That would allow this protocol to be WASM-friendly.

I don't think we should introduce any complexity for the sake of supporting WASM at this point. It is not a critical protocol. I don't see it used in WASM any time soon.

> Plus, in the spirit of https://github.com/libp2p/rust-libp2p/issues/3111, it would be consistent with not introduce binaries in our protocol crates.

Consistency is a good argument in my eyes.

Reply:
> Which folder should these binaries go into?

You'd have to make another crate, like `interop-tests`. Perhaps `perf-utils`, `perf-cs` (for perf-client-server) or `perf-bins`?

Reply:
You could put them into `misc/` too?

Reply:
@thomaseizinger do you feel strongly about this? Thinking about it some more, I don't think the split into two folders is worth the effort at this point. I expect this crate to change quite a bit after this pull request merges.

Reply:
Sounds good.